File: examples/metal/reserved_ip_block/example_1/python/__main__.py
import pulumi
import pulumi_equinix as equinix

two_elastic_addresses = equinix.metal.ReservedIpBlock("twoElasticAddresses",
    project_id=project_id,
    metro="sv",
    quantity=2)
test1 = equinix.metal.ReservedIpBlock("test1",
    project_id=project_id,
    type=equinix.metal.IpBlockType.PUBLIC_I_PV4,
    metro="sv",
    quantity=1)
test = equinix.metal.ReservedIpBlock("test",
    project_id=project_id,
    type=equinix.metal.IpBlockType.GLOBAL_I_PV4,
    quantity=1)

File: examples/metal/reserved_ip_block/example_1/go/main.go
package main

import (
	"github.com/equinix/pulumi-equinix/sdk/go/equinix/metal"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := metal.NewReservedIpBlock(ctx, "twoElasticAddresses", &metal.ReservedIpBlockArgs{
			ProjectId: pulumi.Any(projectId),
			Metro:     pulumi.String("sv"),
			Quantity:  pulumi.Int(2),
		})
		if err != nil {
			return err
		}
		_, err = metal.NewReservedIpBlock(ctx, "test1", &metal.ReservedIpBlockArgs{
			ProjectId: pulumi.Any(projectId),
			Type:      pulumi.String(metal.IpBlockTypePublicIPv4),
			Metro:     pulumi.String("sv"),
			Quantity:  pulumi.Int(1),
		})
		if err != nil {
			return err
		}
		_, err = metal.NewReservedIpBlock(ctx, "test", &metal.ReservedIpBlockArgs{
			ProjectId: pulumi.Any(projectId),
			Type:      pulumi.String(metal.IpBlockTypeGlobalIPv4),
			Quantity:  pulumi.Int(1),
		})
		if err != nil {
			return err
		}
		return nil
	})
}
